contrib/unit_test_coverage/generate_coverage_report.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 The Orbit Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# --- generate_coverage_report.sh ---
+#
+# This script generates a coverage report for the Orbit repository. It uses 
+# llvm-cov and requires a build with build arguments -fprofile-instr-generate 
+# and -fcoverage-mapping. 
+#
+# usage: generate_coverage_report.sh SOURCE_DIR BUILD_DIR OUTPUT_DIR
+#
+# settings:
+# grep inverted matching list for files in BUILD_DIR/bin/ that are not used
+BIN_FILE_FILTER="\.debug\|crashpad_handler"
+# grep inverted matching list for source files that are not used
+SOURCE_FILE_FILTER="Test.cpp\|Tests.cpp\|Fuzzer.cpp"
+# ------------------------------------
+
+# fail on any error
+set -e
+
+function get_css_class_string {
+  local percent=$1
+  local result="column-entry-red"
+  if (( $(echo "$percent > 80" | bc) )); then
+    result="column-entry-yellow"
+  fi
+  if (( $(echo "$percent >= 100" | bc) )); then
+    result="column-entry-green"
+  fi
+  echo "$result"
+}
+
+function generate_html_td {
+  local count=$1
+  local covered=$2
+  local percent=$(jq -n "$covered / $count * 100")
+  local percent_rounded=$(printf "%0.2f" $percent)
+  local css_class=$(get_css_class_string $percent_rounded)
+  local percent_formatted=$(printf "%6s" "$percent_rounded")
+  echo "<td class='$css_class'><pre> $percent_formatted% ($covered/$count)</pre></td>"
+}
+
+if [ "$#" -ne 3 ]; then
+  echo "error: wrong number of arguments."
+  echo "usage: generate_coverage_report.sh SOURCE_DIR BUILD_DIR OUTPUT_DIR"
+  exit 1
+fi
+
+SOURCE_DIR=$(realpath $1)
+if [ ! -d $SOURCE_DIR ]; then
+  echo "error: directory $1 does not exist."
+  exit 1
+fi
+
+BUILD_DIR=$(realpath $2)
+if [ ! -d $BUILD_DIR ]; then
+  echo "error: directory $2 does not exist."
+  exit 1
+fi
+
+OUTPUT_DIR=$(realpath $3)
+if [ ! -d $OUTPUT_DIR ]; then
```

Comment:
I would even argue that we might wanna have that check the other way around. If that script is generating more than one file I would prefer that the script fails when I accidentally pass in an existing directory, to avoid unintended overwrites or that an existing directory is cluttered with unrelated files. WDYT?

Reply:
Kind of agree, but this depends a bit on two things:
1. how does this integrate with the usage scenario we have.
2. if we only add additional data, it might be fine, to have the directory existing.

Reply:
Good point. The more I'm thinking about that the less I'm sure what to do here. In general I would consider this script somehow work-in-progress. Something we can iterate on.

My vision for the developer would be to have a single command. Something like `./build.sh coverage` which builds Orbit with the correct flags, runs the tests, and also generates the coverage report. But this means cmake needs to call that script. Before having looked into that I don't know what is the "right" thing to do for the script. So I would argue we can also change that later.

So Anton from my side feel free to take any decision on this. :sweat_smile:

Reply:
I am also not 100% on this aka this will be changed when necessary, but I decided to go a completely different way, so please let me know what you think. 

The output directory is now always BUILD_DIR/coverage. I also added a echo at the end of the script which tells the user where to find the report

Reply:
Seems legit. Thanks